File: app/src/main/java/com/angogasapps/myfamily/ui/screens/buy_list/dialogs/AddBuyListDialog.java
package com.angogasapps.myfamily.ui.screens.buy_list.dialogs;

import android.content.Context;
import android.widget.EditText;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;

import com.angogasapps.myfamily.R;
import com.angogasapps.myfamily.firebase.BuyListFunks;
import com.angogasapps.myfamily.firebase.interfaces.IOnEndCommunicationWithFirebase;
import com.angogasapps.myfamily.models.BuyList;

import es.dmoral.toasty.Toasty;

public class AddBuyListDialog {
    private Context context;
    private BuyList buyList;


    public AddBuyListDialog(@NonNull Context context) {
        this.context = context;
    }

    public AddBuyListDialog(Context context, BuyList buyList){
        this(context);
        this.buyList = buyList;
    }

    public void show(){
        AlertDialog.Builder alertDialog = new AlertDialog.Builder(context);


        alertDialog.setTitle(context.getString(R.string.enter_buy_list_name));


        final EditText input = new EditText(context);
        LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT,
                LinearLayout.LayoutParams.MATCH_PARENT);
        input.setLayoutParams(lp);

        if (buyList != null){
            input.setText(buyList.getName());
        }
        alertDialog.setView(input);

        alertDialog.setPositiveButton(context.getString(R.string.cancel),
                (dialog, which) -> {
                    dialog.cancel();
                });

        alertDialog.setNegativeButton(context.getString(R.string.add),
                (dialog, which) -> {
                    String str = input.getText().toString().trim();
                    if (str != null && !str.equals("")) {
                        BuyList inputBuyList;
                        if (buyList ==null) {
                            inputBuyList = new BuyList(str);
                            BuyListFunks.addNewBuyList(inputBuyList, new IOnEndCommunicationWithFirebase() {
                                @Override
                                public void onSuccess() {}
                                @Override
                                public void onFailure() {}
                            });
                        }else{
                            inputBuyList = new BuyList(buyList);
                            inputBuyList.setName(str);
                            BuyListFunks.updateBuyListName(inputBuyList, new IOnEndCommunicationWithFirebase() {
                                @Override
                                public void onSuccess() {}
                                @Override
                                public void onFailure() {}
                            });
                        }
                    }else{
                        Toasty.error(context, R.string.enter_buy_list_name).show();
                    }
                });
        alertDialog.show();
    }
}

File: app/src/main/java/com/angogasapps/myfamily/ui/screens/buy_list/dialogs/AddBuyListDialog.kt
package com.angogasapps.myfamily.ui.screens.buy_list.dialogs

import android.content.Context
import android.content.DialogInterface
import com.angogasapps.myfamily.models.buy_list.BuyList
import com.angogasapps.myfamily.R
import android.widget.EditText
import android.widget.LinearLayout
import androidx.appcompat.app.AlertDialog
import com.angogasapps.myfamily.app.appComponent
import com.angogasapps.myfamily.network.repositories.BuyListRepository
import es.dmoral.toasty.Toasty
import javax.inject.Inject

class AddBuyListDialog(private val context: Context) {
    private var buyList: BuyList? = null
    @Inject
    lateinit var buyListRepository: BuyListRepository

    init {
        appComponent.inject(this)
    }

    constructor(context: Context, buyList: BuyList?) : this(context) {
        this.buyList = buyList
    }

    fun show() {
        val alertDialog = AlertDialog.Builder(
            context
        )
        alertDialog.setTitle(context.getString(R.string.enter_buy_list_name))
        val input = EditText(context)
        val lp = LinearLayout.LayoutParams(
            LinearLayout.LayoutParams.MATCH_PARENT,
            LinearLayout.LayoutParams.MATCH_PARENT
        )
        input.layoutParams = lp
        if (buyList != null) {
            input.setText(buyList!!.title)
        }
        alertDialog.setView(input)
        alertDialog.setPositiveButton(
            context.getString(R.string.cancel)
        ) { dialog: DialogInterface, which: Int -> dialog.cancel() }
        alertDialog.setNegativeButton(
            context.getString(if (buyList == null) R.string.add else R.string.change)
        ) { dialog: DialogInterface?, which: Int ->
            val name = input.text.toString().trim { it <= ' ' }
            if (name != "") {
                val inputBuyList: BuyList
                if (buyList == null) {
                    inputBuyList = BuyList(title = name)
                    buyListRepository.createNewBuyList(inputBuyList)
                } else {
                    inputBuyList = BuyList(buyList!!)
                    inputBuyList.title = name
                    buyListRepository.updateBuyListName(inputBuyList)
                }
            } else {
                Toasty.error(context, R.string.enter_buy_list_name).show()
            }
        }
        alertDialog.show()
    }
}